### PR TITLE
[ENH] Preview of the data dictionary

### LINKED ISSUE #415: Preview of the data dictionary

### PR DESCRIPTION
**Pull request by Manya-tech**
- Closes #415

Changes proposed in this pull request:

- Displayed the .json data dictionary on the downloads page after formatting it

Issues:

- Tried to display the data dictionary inside a box for better UI but encountered errors while trying to commit those changes.

Intended result:
image
image

Actual result you will see:
image
image

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: pages/download.vue
Comment (surchs) [inline pages/download.vue:39]: Right now, this is always visible. That'd be confusing if the data dictionary is not valid and therefore nothing is shown in the next block.

**Review by surchs — CHANGES_REQUESTED** [review]
Thanks @Manya-tech for taking this on, and welcome to our repository :tada:!

I have some suggestions / requests before we can merge this:
- data dictionaries can be pretty long and clutter the page, could you wrap the preview section in a collapsible element so the user can chose when to look at it? We have some examples of this on the `Home.vue` page I think, but something like https://bootstrap-vue.org/docs/components/collapse should do it
- could you also ensure that the `pre` block is scrollable (again, long files are tricky to look at)
- finally: any of these UI elements should be consistent with the preview. That is: we should not have a button to "preview data dictionary" and then not show anything, e.g. because the data dictionary is not valid. 

I don't have a strong preference for whether we should or should not show an invalid data dictionary. Leaning slightly towards "only show when valid". @rmanaem wdyt?

**Comment by rmanaem** [comment]
> I don't have a strong preference for whether we should or should not show an invalid data dictionary. Leaning slightly towards "only show when valid". @rmanaem wdyt?

I think displaying an invalid data dictionary on the download page would be helpful as a visual reference for users when they encounter an error.

I also think it's a good idea to have a test for this preview. Let me know what you think @surchs.

**Comment by Manya-tech** [comment]
Thank you @surchs for the warm welcome. I really like this project and am excited to contribute.

I will wrap the preview section in the collapsible element. I tried to make the pre block scrollable but I was not able to commit the code for that. I was facing some errors, but I am trying to solve that. 

Currently, the code I wrote displays only the valid .json data dictionary. If the invalid data dictionary should also be displayed, is there something I should keep in mind?

**Comment by Manya-tech** [comment]
Hello @surchs and @rmanaem ,

I've completed all the tasks requested by @surchs. The preview section has been updated to include a collapsible element and is now scrollable. Additionally, the code for handling invalid data dictionaries is also present in the file. 

I am attaching the image showcasing the features I have added.

image

File: pages/download.vue
Comment (Manya-tech) [reply pages/download.vue:39]: The code now handles valid and invalid data dictionaries properly. If the data dictionary is invalid, it notifies the user and offers appropriate information.

Comment (surchs) [reply pages/download.vue:39]: awesome, thank you very much @Manya-tech! Do you feel comfortable to add a test for this new functionality you added? The test should probably live here: https://github.com/neurobagel/annotation_tool/blob/main/cypress/e2e/page/download-pagetests.cy.js

If no, we can also add the test for you by taking over the PR.

File: pages/download.vue
Comment (surchs) [inline pages/download.vue:36]: This section still shows up every time. I think we would want it to only show up if the dictionary is valid. A test could be useful to check this.

**Comment by surchs** [comment]
Thanks @Manya-tech for working on this! I think the PR is almost there. I left two more comments, let me know if you have time to implement these. Otherwise we might take over the PR and wrap it up

**Comment by Manya-tech** [comment]
I would like to implement these. Could you please help me if I have any doubts about adding tests?

**Comment by rmanaem** [comment]
@Manya-tech Absolutely, feel free to ping either of us if you have any questions.

**Comment by Manya-tech** [comment]
I have added a test for the preview. @surchs mentioned that a particular section should only show up if the data dictionary is valid, but @rmanaem suggested that displaying an invalid data dictionary on the download page would be helpful as a visual reference for users when they encounter an error. So, I am a bit confused.

If the data dictionary is invalid, the "incomplete annotations" message is displayed. Otherwise, the "congratulations" message is displayed. Users can then review their data dictionary and download or force download it.

I am adding screenshots for both cases. Please let me know what changes I should make.

image

image

**Review by surchs — APPROVED** [review]
Awesome @Manya-tech, thanks a lot for this PR! This all looks good to me. I'll wait for @rmanaem to take a look and then we can merge this.

@rmanaem: I think we can release this. Will apply the label, wdyt?

**Pull request merged**
